### ISSUE TITLE
Hardware: Have a single board computer spare? Will it run PiNode-XMR?

### ISSUE THREAD

**Issue by shermand100** [shermand100/pinode-xmr#17]
## I'd like some assistance from the community!

With the self installer scripts releasing us from the constraints of the Raspberry Pi platform I require testers with access to single board computers (SBC). It would be rather wasteful (and expensive) for me to buy one of every SBC out there so I'm asking you to dig though your spare project box's to see what you can put to use.

The list below is not exhaustive. If you own something else that's not on the list, that's fine, give it a go, it can be added. Use the list as a guide for the info we need. If you use a non-standard disk image can you link where others can find it, however standard images should be used where possible as the point of this open source installer is that it can be trusted.

Leave your findings as a comment at the bottom.

**My Expectations:**  The purpose of this is to see what variations (if any) affect the PiNode-XMR installer or how it functions once complete. I believe that on each device the node will install and be functional if using Debian Buster, but variations in hardware such as CPU temperature sensors may become inaccurate. Details like this will be logged in the notes column, or a link to their own issue as they arise.

| Device          | Base disk Image | Tested?|   Status  | Build time        | Notes |
| :--------------:| :-------------: |:------:| :------:  |:-----------------:|:-----:|
| Pine A64-LTS    | [buster_current](https://dl.armbian.com/pine64/)  | PENDING| ❓        |❓                 |     |
| Rock 64 | [buster_current](https://dl.armbian.com/rock64/)  | PENDING| ❓        |❓                 |     |
| Le Potato| [buster_current_minimal](https://dl.armbian.com/lepotato/)  | PENDING| ❓        |❓                 |     |
| Banana Pi M64| [buster_current_minimal](https://dl.armbian.com/bananapim64/)  | PENDING| ❓        |❓                 |     |
| Odroid C2 | [buster_current_minimal](https://dl.armbian.com/odroidc2/)  | PENDING| ❓        |❓                 |     |
| Rock Pi 4B| [buster_current](https://dl.armbian.com/rockpi-4b/)  | PENDING| ❓        |❓                 |     |
| NanoPC-T3 Plus| [buster](https://dl.armbian.com/nanopct3plus/)  | PENDING| ❓        |❓                 | Legacy base images listed    |
|Asus Tinkerboard| [buster_current_minimal](https://dl.armbian.com/tinkerboard/)  | PENDING| ❓        |❓                 |     |

### Confirmed

| Device          | Base disk Image | Tested?|   Status  | Build time        | Notes |
| :--------------:| :-------------: |:------:| :------:  |:-----------------:|:-----:|
| Raspberry Pi 3b+| [Raspbian Buster lite 2020-02-13](https://www.raspberrypi.org/downloads/raspbian/)   | ✅ | ✅ |Approx 8hrs|        |
| Raspberry Pi 4  | [Raspbian Buster lite 2020-02-13](https://www.raspberrypi.org/downloads/raspbian/)   | ✅ | ✅ |Approx 4hrs|         |
| Odroid XU4      | [Buster_current_5.4.28_minimal](https://dl.armbian.com/odroidxu4/) | ✅ | ✅ |Approx 4hrs|    |
| Orange Pi Plus 2E| [buster_current](https://dl.armbian.com/orangepiplus2e/)  | ✅  | ✅ | Approx 4-6 hours  | Community tested by @dalecooper  |

**Comment by dalecooper** [comment]
I can confirm it installs on Orange Pi + 2E with absolutely no problem. I can't tell how long it takes to build because I've let it work overnight but I guess it's between a Pi 3 and a Pi 4. Maybe there are some dated logs somewhere I don't know about ?
RPC Pay and mining won't start, but I'm pretty sure it's not device related. Will investigate further when I have more time, and test more features too.
But so far I'm happy with my private full node.

**Comment by shermand100** [comment]
> I can confirm it installs on Orange Pi + 2E with absolutely no problem. 

Thanks so much for your help with this! I can credit your username as the tester in the notes column if you'd like? If so, github or reddit username?

I can't think of a way to derive a build time, but it's not really important. Just a guide for the curious, I'll ball-park between 4 - 7.

Yes the RPC pay won't work just yet. Waiting for a merge from the Monero team of this issue https://github.com/monero-project/monero/pull/6260 which started being introduced last week. The exact issue PiNode-XMR has is this one https://github.com/monero-project/monero/issues/3083

Mining should work though (but be terrible) I'll take a look at it.
Enjoy your node 

**Edit:**  Good spot on the Mining issue btw. Had a broken link to a variable for in-peers it needed. I think I killed it when I had a tidy up a few weeks ago. Fixed now though for future builders or if you force an update.
Getting a whole <1 H/s with my Pi3 (No AES support for RandomX with most single board computers).
